lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -6234,9 +6194,8 @@ static bool isSelfDerivedNestedTypeNameMatchEdge(
 
   DependentMemberType *sourceDepMemTy;
   while ((sourceDepMemTy = sourceType->getAs<DependentMemberType>()) &&
-         target->getParent() &&
-         sourceDepMemTy->getAssocType() &&
-         sourceDepMemTy->getAssocType() == target->getResolvedAssociatedType()){
+         target->getParent() && sourceDepMemTy->getAssocType() &&
```

Comment:
Oh, I don't think you need to check for a null `sourceDepMemTy->getAssocType()`, because the "subject" of any constraint should (now) always be a well-formed type. I bet there are other places where we have such null checks that could go away with your PR!

Reply:
Right, good catch! Thanks!